silq/analysis/fit_toolbox.py

```diff
@@ -43,6 +46,38 @@ def __init__(self, fit=True, print=False, plot=None, **kwargs):
             if fit:
                 self.perform_fit(print=print, plot=plot, **kwargs)
 
+    def __getitem__(self, item) -> float:
+        """Retrieve fitted parameter value"""
+        return self.fit_result.best_values[item]
```

Comment:
If I do something like this:
```python
fit = SineFit()
fit['frequency']
```
I get the following error
```python-traceback
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-18-827926a6d6a1> in <module>
      1 fit = SineFit()
      2 
----> 3 fit['frequency']

~/Documents/github/SilQ/silq/analysis/fit_toolbox.py in __getitem__(self, item)
     49     def __getitem__(self, item) -> float:
     50         """Retrieve fitted parameter value"""
---> 51         return self.fit_result.best_values[item]
     52 
     53     def __call__(

AttributeError: 'NoneType' object has no attribute 'best_values'
```

This case is quite trivial to figure out why it failed, but I think this should be captured and a more informative error given.

e.g. 
```suggestion
    def __getitem__(self, item) -> float:
        """Retrieve fitted parameter value"""
        if self.fit is not None:
             return self.fit_result.best_values[item]
        else:
              raise RuntimeError("No fit result to get parameters from.") # Probably not a key or index error, but something more informative than this would be best
```

Reply:
Done (without the comment)